symphony-bdk-core/src/test/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedServiceV2Test.java

```diff
@@ -0,0 +1,303 @@
+
+        assertThrows(ApiException.class, this.datafeedService::start);
+        verify(datafeedApi, times(1)).listDatafeed("1234", "1234");
+        verify(datafeedApi, times(2)).createDatafeed("1234", "1234");
+    }
+
+    @Test
+    void testStartClientErrorReadDatafeed() throws ApiException, AuthUnauthorizedException {
+        AckId ackId = datafeedService.getAckId();
+        when(datafeedApi.listDatafeed("1234", "1234")).thenReturn(Collections.singletonList(new V5Datafeed().id("test-id")));
+        when(datafeedApi.createDatafeed("1234", "1234")).thenReturn(new V5Datafeed().id("recreate-df-id"));
+        when(datafeedApi.readDatafeed("test-id", "1234", "1234", ackId)).thenThrow(new ApiException(400, "client-error"));
+        when(datafeedApi.readDatafeed("recreate-df-id", "1234", "1234", ackId))
+                .thenReturn(new V5EventList().addEventsItem(new V4Event().type(DatafeedEventConstant.MESSAGESENT)).ackId("ack-id"));
+
+        this.datafeedService.start();
+        verify(datafeedApi, times(1)).listDatafeed("1234", "1234");
+        verify(datafeedApi, times(1)).readDatafeed("test-id", "1234", "1234", ackId);
+        verify(datafeedApi, times(1)).readDatafeed("recreate-df-id", "1234", "1234", ackId);
+        verify(datafeedApi, times(1)).createDatafeed("1234", "1234");
+        verify(datafeedApi, times(1)).deleteDatafeed("test-id", "1234", "1234");
+    }
+
+    @Test
+    void testStartAuthErrorReadDatafeed() throws ApiException, AuthUnauthorizedException {
+        AckId ackId = datafeedService.getAckId();
+        when(datafeedApi.listDatafeed("1234", "1234")).thenReturn(Collections.singletonList(new V5Datafeed().id("test-id")));
+        when(datafeedApi.readDatafeed("test-id", "1234", "1234", ackId)).thenThrow(new ApiException(401, "client-error"));
+        doThrow(AuthUnauthorizedException.class).when(authSession).refresh();
+
+        assertThrows(AuthUnauthorizedException.class, this.datafeedService::start);
+        verify(datafeedApi, times(1)).listDatafeed("1234", "1234");
+        verify(datafeedApi, times(1)).readDatafeed("test-id", "1234", "1234", ackId);
+        verify(authSession, times(1)).refresh();
+    }
+
+    @Test
+    void testStartServerErrorReadDatafeed() throws ApiException {
+        AckId ackId = datafeedService.getAckId();
+        when(datafeedApi.listDatafeed("1234", "1234")).thenReturn(Collections.singletonList(new V5Datafeed().id("test-id")));
+        when(datafeedApi.readDatafeed("test-id", "1234", "1234", ackId)).thenThrow(new ApiException(500, "client-error"));
+
+        assertThrows(ApiException.class, this.datafeedService::start);
+        verify(datafeedApi, times(1)).listDatafeed("1234", "1234");
+        verify(datafeedApi, times(2)).readDatafeed("test-id", "1234", "1234", ackId);
+    }
+
+    @Test
+    void testStartClientErrorDeleteDatafeed() throws ApiException, AuthUnauthorizedException {
```

Comment:
running this test locally shows that the retry is used and at some points it seems to wait for 500ms
but should it be the case for this scenario? we have an invalid DF if, recreate one and wait 500ms before it works again?